isotovideo

```diff
@@ -135,17 +135,14 @@ sub _exit_code_from_test_results () {
     my @results = glob(path(bmwqemu::result_dir(), "result-*.json"));
     return EXIT_STATUS_ERR_NO_TESTS if @results == 0;
 
-    my $did_fail = 0;
     for my $result_file_path (@results) {
         my $result_file = path($result_file_path);
         my $test_result = decode_json($result_file->slurp)->{result};
         diag sprintf("Test result [%s] %s", $result_file->to_string, $test_result);
-        # If a failure (anything different from ok & softfail) is found, keep it.
-        next if $did_fail;
-
-        $did_fail = $test_result !~ m/^(ok|softfail)$/;
+        # If a failure (anything different from ok & softfail) is found, return immediately the exit status.
+        return EXIT_STATUS_ERR_FROM_TEST_RESULTS if $test_result !~ m/^(ok|softfail)$/;
     }
-    return $did_fail ? EXIT_STATUS_ERR_FROM_TEST_RESULTS : EXIT_STATUS_OK;
```

Comment:
I'd be more careful about this change. Why was it introduced like this originally then to distinguish between the cases instead of immediately returning on failure? Maybe this has an impact on using "ignore_failure" from http://open.qa/docs/#_test_flags ?

Reply:
As far as I can tell this has been moved without changing how it works in the latest revision.